### PR TITLE
Indexed organization events into ElasticSearch

### PR DESCRIPTION
**Pull request by umbreak**

### CODE REVIEW

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/organizations/OrganizationEsIndexer.scala [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+  private implicit val orgIdQualifier: ConfiguredQualifier[OrgId]   = Qualifier.configured[OrgId](base)
+  private implicit val stringQualifier: ConfiguredQualifier[String] = Qualifier.configured[String](baseVoc)
+
+  private val revKey               = "rev".qualifyAsString
+  private val deprecatedKey        = "deprecated".qualifyAsString
+  private val orgName              = "name".qualifyAsString
+  private lazy val indexJson: Json = jsonContentOf("/es-index.json", Map(quote("{{type}}") -> t))
+
+  private def createIndexIfNotExist(id: OrgId): F[Unit] = {
```
Comment (bogdanromanx) [inline modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/organizations/OrganizationEsIndexer.scala]: this needs to be cached as we can't afford checking the index for every event; an alternative would be to attempt to create a document and recover with index creation.

Comment (umbreak) [reply modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/organizations/OrganizationEsIndexer.scala]: Yes I agree. I was gonna do it with a `Map`.

**Comment by umbreak** [comment]
Jenkins retest this please

**Comment by umbreak** [comment]
Jenkins retest this please

**Comment by umbreak** [comment]
Jenkins retest this please

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/organizations/OrganizationEsIndexer.scala [hunk 2]
```diff
@@ -50,13 +51,17 @@ class OrganizationEsIndexer[F[_]](client: ElasticClient[F],
   private val deprecatedKey        = "deprecated".qualifyAsString
   private val orgName              = "name".qualifyAsString
   private lazy val indexJson: Json = jsonContentOf("/es-index.json", Map(quote("{{type}}") -> t))
+  private val indices              = Set[String]()
```
Comment (bogdanromanx) [inline modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/organizations/OrganizationEsIndexer.scala]: This data structure is not thread safe, which could result in arbitrary failures in concurrent cases. Are concurrent cases possible of all indexer types (org / domain / schema / context / instance)?

Comment (bogdanromanx) [reply modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/organizations/OrganizationEsIndexer.scala]: I assume we don't, but could you confirm?

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/ConcurrentSet.scala [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  /**
+    * Removes a single element from the set.
+    *
+    * @param elem the element
+    */
+  def -=(elem: A): Set[A] = set -= elem
+
+  private val set: Set[A] =
+    newSetFromMap(new ConcurrentHashMap[A, lang.Boolean](elems.map(k => k -> lang.Boolean.TRUE).toMap.asJava)).asScala
```
Comment (bogdanromanx) [inline modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/ConcurrentSet.scala]: As described here: https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#newSetFromMap-java.util.Map-
... the map must be empty when wrapped in a set.

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/collection/ConcurrentSerBuilder.scala [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package ch.epfl.bluebrain.nexus.kg.core.collection
+
+import java.lang
+import java.util.Collections.newSetFromMap
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.Set
+
+object ConcurrentSerBuilder {
```
Comment (bogdanromanx) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/collection/ConcurrentSerBuilder.scala]: typo

**Pull request merged**